src/python/pants/backend/visibility/rules.py

```diff
@@ -41,7 +42,11 @@ def build_file_visibility_implementation(
 @rule
 async def visibility_validate_dependencies(
     request: VisibilityValidateDependenciesRequest,
+    visibility: VisibilitySubsystem,
 ) -> ValidatedDependencies:
+    if not visibility.enforce:  # TODO: support for `or current_goal == "lint"`
```

Comment:
any way to determine the current goal (or preferably, lint tool if any) executing?

Reply:
#17674

Reply:
#18788